main.c

```diff
@@ -321,7 +321,14 @@ static int handle_property_changed(sd_bus_message *msg, void *userdata,
 		}
 
 		const char *prop;
-		while ((ret = sd_bus_message_enter_container(msg, 'e', "sv")) > 0) {
+		while (true) {
+			ret = sd_bus_message_enter_container(msg, 'e', "sv");
+			if (ret < 0) {
+				if (-ret == ENXIO) {
+					break;
+				}
+				goto error;
+			}
```

Comment:
I'm not sure this is the best way to address this. Clang-tidy warned that the `ret` that is assigned in the `while` loop condition is not used. I chose to make this explicit check for ENXIO to indicate the end of the message instead of considering any error to indicate that like previously. 
But why I think this may not necessarily be the best way to address this is that the possible ENXIO error is not documented for that function. However, after quick search it seemed that the documentation for sd_bus is not as good as for example for swayidle.

Reply:
> But why I think this may not necessarily be the best way to address this is that the possible ENXIO error is not documented for that function.

Hm, but shouldn't `sd_bus_message_enter_container` return 0 in this case?

Reply:
I debugged this a bit more and found out that when the message is processed correctly, `sd_bus_message_enter_container` will return 0 instead of -6 at the end of the container. But the `sd_bus_message_skip` is called with wrong type argument currently causing it to return -6 and `sd_bus_message_exit_container` is not called at the end of the loop while it should be.

I'll make a separate PR fixing that message handling properly and remove this ENXIO handling commit from this PR as it is not the correct thing to do.

Reply:
I opened #96 to address this message parsing issue. I'll clean up this PR when I find the next free time slot.

Reply:
Nice find!

Reply:
Rebased this branch on top of master with --autosquash and dropped the commit with this bad change now that it got properly addressed in that other PR.